pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -277,14 +278,13 @@ func (c *Controller) syncOwnedProvisionRequest(
 			expectedPodSets := requiredPodSets(wl.Spec.PodSets, prc.Spec.ManagedResources)
 			psaMap := slices.ToRefMap(wl.Status.Admission.PodSetAssignments, func(p *kueue.PodSetAssignment) kueue.PodSetReference { return p.Name })
 			podSetMap := slices.ToRefMap(wl.Spec.PodSets, func(ps *kueue.PodSet) kueue.PodSetReference { return ps.Name })
-			for _, psName := range expectedPodSets {
-				ps, psFound := podSetMap[psName]
-				psa, psaFound := psaMap[psName]
-				if !psFound || !psaFound {
-					return errInconsistentPodSetAssignments
-				}
+			mergedPodSets, err := mergePodSets(expectedPodSets, podSetMap, psaMap, prc.Spec.MergePodSets)
```

Comment:
What if the identity keys are used during the AdmissionCheck `Retry` something waiting phase?
Let's say the PodSet has gpu tolerations on creation, after that the PodSet tolerations are removed during AC `Retry` phase.

Reply:
ACs should not be modifying the PodSet template directly. This is what the user supplied. If some AC want to make adjustments then the PodSetUpdates in the workload.status admissionChecks.podSetUpdates should be used. I think the identity check should only care about equivalence of what is supplied to the ClusterAutoscaler, so what is in PosSet PodTemplate directly. ignoring podSetUpdates.

Reply:
Yes, updating the Workload is what I meant. I don't point to manually updated ProvisioningRequest.
I wanted to check if we can handle those appropriately, even if users update Workload with a container spec, something.

Reply:
We don't support in Kueue users updating the container specs inside the workload.status.podSets. 

If users update workload.status.podSets they may run into issues (like immediate Workload eviction),  but I don't think there is any new class of issues they may run into by merging of PodSets.

Reply:
Thank you for describing that. If there are any conflicts when recomputing the merged PodSet, I'm ok with that.